waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -130,6 +139,361 @@ protected String getUsage() {
+         *
+         * @param type Type of validation
+         * @param results List containing the validation results for each partition
+         * @return True if no error in validation. False otherwise
+         */
+        private boolean verifyValidation(ValidationResults.ValidationType type, List<PartitionValidationResults> results) {
+            boolean success = true;
+            for (int partitionId = 0; partitionId < results.size(); partitionId++) {
+                if (!verifyValidation(type, results, partitionId)) {
+                    success = false;
+                }
+            }
+            return success;
+        }
+
+        /**
+         * Verify a validation type for a specific partition from the list of validation results.
+         * If validation failed, error is printed
+         *
+         * @param type Type of validation
+         * @param results List containing the validation results for each partition
+         * @param partitionId Partition to validate
+         * @return True if no error in validation. False otherwise
+         */
+        private boolean verifyValidation(ValidationResults.ValidationType type, List<PartitionValidationResults> results, int partitionId) {
+            PartitionValidationResults partitionResult = results.get(partitionId);
+
+            ValidationResults partitionZkResults = partitionResult.validationResultsMap.get(type);
+            if (partitionZkResults.status.equals(ValidationResults.Status.FAILURE)) {
+                System.out.println("Validation " + type.name() + " failed for partition " + partitionId);
+                System.out.println("Error is: " + partitionZkResults.error);
+                return false;
+            }
+            return true;
+        }
+
+        /**
+         * Validate for a single server in the cluster the consistency of partition assignments on the actual server
+         * versus on Zookeeper metadata.
+         *
+         * @param rpcClient Client used to connect to Waltz server to fetch partition assignments
+         * @param server Actual server to run the validation for
+         * @param clusterManager Contains zkclient used to fetch partition assignments of Zookeeper
+         * @param partitionValidationResultsList List indexed by partition in which validation outputs are inserted
+         * @return Completable future that complete once the partitionValidationResultsList is filled with validation data
+         *          from all partitions from that server
+         * @throws InterruptedException If thread interrupted while waiting for channel with Waltz server to be ready
+         * @throws ClusterManagerException Thrown if Zookeeper is missing some ZNodes or ZNode values
+         */
+        private CompletableFuture<Void> buildServerZKPartitionAssignmentsValidation(InternalRpcClient rpcClient,
+                                                                                    ServerDescriptor server,
```

Comment:
should we rename thsi to `serverDescriptor` instead of `server`?? because ServerDescriptor only contains the information of the Server. It's not the actual server. I feel like this might be confusing.

Reply:
fixed. Thanks!